ckan/controllers/user.py

```diff
@@ -162,10 +166,90 @@ def register(self, data=None, errors=None, error_summary=None):
 
         return self.new(data, errors, error_summary)
 
+    def _generate_internal_logs(self, log_message):
+        '''
+        This helper function is called if sending a new user registration email fails.
+        '''
+
+        logging.warning("There was an error sending the email. Writing to log file.")
+
+        os.chdir(os.environ['REGISTRATION_ERROR_LOG_FILE_PATH'])
+
+        with open(os.environ['REGISTRATION_ERROR_LOG_FILE_NAME'], "a") as fp:
```

Comment:
stick a `try... except` around this so that if that file can't be opened the app doesn't crater

Reply:
Fixed.

Reply:
Fixed.